Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverPool.hs

```diff
@@ -312,6 +317,22 @@ getCurrentWindowAvailability ::
   m [Maybe a]
 getCurrentWindowAvailability merchantId driverId = Redis.withCrossAppRedis . withAvailabilityTimeWindowOption merchantId $ SWC.getCurrentWindowValues (mkAvailableTimeKey driverId.getId)
 
+aquireLockOnDriverForSearch ::
+  ( Redis.HedisFlow m r,
+    CacheFlow m r
+  ) =>
+  Id DM.Merchant ->
+  Id DP.Driver ->
+  Double ->
+  Int ->
+  Int ->
+  m Bool
+aquireLockOnDriverForSearch merchantId driverId time n expirationTime = do
+  HQ.eval
+    "local validRequest = {} ;local lengthvalidRequest=0;for index,value in ipairs(redis.call('LRANGE', tostring(KEYS[1]),0,-1)) do if tonumber(value) >= tonumber(ARGV[1]) then table.insert(validRequest,tostring(value));lengthvalidRequest = lengthvalidRequest + 1;end; end; if lengthvalidRequest == 0 then validRequest = {tostring(ARGV[1])};lengthvalidRequest=1;end;redis.call('DEL',tostring(KEYS[1]));redis.call('LPUSH',tostring(KEYS[1]),unpack(validRequest));redis.call('EXPIRE',tostring(KEYS[1]),tonumber(ARGV[3])); return lengthvalidRequest < tonumber(ARGV[2]);"
+    [mkParallelSearchRequestKey merchantId driverId]
+    [show time, show n, show expirationTime]
```

Comment:
use, `time` + `expirationTime ` as the value to set inside lua script for validTill.